cmd/snap-confine/ns-support.c

```diff
@@ -486,7 +487,7 @@ static int sc_inspect_and_maybe_discard_stale_ns(int mnt_fd,
 		debug("preserved mount is not stale, reusing");
 		return 0;
 	case SC_DISCARD_SHOULD:
-		if (sc_cgroup_freezer_occupied(inv->snap_instance)) {
+                if (!sc_cgroup_is_v2() && sc_cgroup_freezer_occupied(inv->snap_instance)) {
```

Comment:
I wonder if sc_cgroup_is_v2() we should always assume it is stale, but occupied? Or trigger SC_DISCARD_MUST. It is hard to know what to do, but my thinking is that perhaps we can at least provide dependable, broken behavior for snaps on v2 only systems, as opposed to the occasional odd error.

Reply:
I think this is a useful compromise on the way towards better cgroup v2 support.

Reply:
Added a code path for this.